### PR TITLE
Closes #4872: Handling of loops for Ansible operators

### LINKED ISSUE #4872: Ansible debug task not working in a loop in ansible-operator
#### What did you do?

Please consider the following debug tasks in:

```yaml
- hosts: localhost
  gather_facts: false
  vars:
    lines:
    - 1
    - 2
    - 3
  tasks:

    - name: Debug without loop
      debug:
        msg: "{{ lines }}"

    - name: Debug with loop
      loop: "{{ lines }}"
      debug:
        msg: "{{ item }}"
```

Running locally, with ansible, the 2 `debug` tasks work and print the lines.

When running in quay.io/operator-framework/ansible-operator:v1.7.0, only the task `Debug without loop` task appears in logs.

Output:

```
--------------------------- Ansible Task StdOut -------------------------------

 TASK [Debug without loop] ********************************
ok: [localhost] => {
    "msg": [
        1,
        2,
        3
    ]
}

-------------------------------------------------------------------------------
{"level":"info","ts":1620205414.6191092,"logger":"logging_event_handler","msg":"[playbook debug]","name":"gpte-agnosticv","namespace":"agnosticv-operator","gvk":"gpte.redhat.com/v1, Kind=AgnosticVRepo","event_type":"runner_on_ok","job":"6129484611666145821","EventData.TaskArgs":""}
```

The  task `Debug with loop` doesn't appear in log. I don't know if it was executed or not.

#### What did you expect to see?

I expect the `debug` task to work with a `loop` too.

#### Workaround

Use `include_tasks` and put the `debug` task in another task file:

```yaml
    - name: Debug with loop + include_tasks
      loop: "{{ lines }}"
      include_tasks: print_error.yaml
```

```yaml
---
- debug:
    msg: "{{ item }}"
```

#### Environment

**Operator type:**  quay.io/operator-framework/ansible-operator:v1.7.0

Also tried: ansible-operator:v1.6.2

/language ansible

**Kubernetes cluster type:** OpenShift 4.6.22

### PR DESCRIPTION
**Pull request by VenkatRamaraju**
Signed-off-by: Venkat Ramaraju <venky2063@gmail.com>

**Description of the change:**
Added checks and formatting to print events that are in a loop while running Ansible operators.

**Motivation for the change:**
To no longer neglect tasks that involve the loop keyword in a task, as mentioned in #4872.

**Note:** My solution doesn't seem like the nicest/easiest one. However, every iteration of the loop comes in as a separate event to `log_events.go`, so this is the workaround I could come up with. Let me know if there's a better way to do it.

### CODE REVIEW

**Comment by VenkatRamaraju** [comment]
For reference, this is how the output looks when the loop keyword is used in a task (After changes from this PR):

Tasks (From the issue):
```
---
- name: Debug without loop
  debug:
    msg: "{{ lines }}"
  vars:
    lines:
      - 1
      - 2
      - 3

- name: Debug with loop
  loop: "{{ lines }}"
  debug:
    msg: "{{ item }}"
  vars:
    lines:
      - 1
      - 2
      - 3
 ```

Ansible Output
```
--------------------------- Ansible Task StdOut -------------------------------

 TASK [Debug without loop] ********************************
ok: [localhost] => {
    "msg": [
        1,
        2,
        3
    ]
}

-------------------------------------------------------------------------------

--------------------------- Ansible Task StdOut -------------------------------

 TASK [Debug with loop] ********************************
ok: [localhost] => (item=1) => {
    "msg": 1
}
ok: [localhost] => (item=2) => {
    "msg": 2
}
ok: [localhost] => (item=3) => {
    "msg": 3
}

-------------------------------------------------------------------------------
```

File: internal/ansible/controller/reconcile.go [hunk 1]
```diff
@@ -170,9 +170,21 @@ func (r *AnsibleOperatorReconciler) Reconcile(ctx context.Context, request recon
 	// iterate events from ansible, looking for the final one
 	statusEvent := eventapi.StatusJobEvent{}
 	failureMessages := eventapi.FailureMessages{}
+	prevEvent := ""
 	for event := range result.Events() {
 		for _, eHandler := range r.EventHandlers {
+			if event.Event == eventapi.EventRunnerItemOnOk && prevEvent != eventapi.EventRunnerItemOnOk { // Loop begins
+				fmt.Printf("\n--------------------------- Ansible Task StdOut -------------------------------\n")
```
Comment (fabianvf) [inline internal/ansible/controller/reconcile.go:177]: Let's keep all the actual logging in the event logging package

Comment (VenkatRamaraju) [reply internal/ansible/controller/reconcile.go:177]: Yeah, I thought the same. But according to this loop, each iteration of the ansible loop gets a separate call to the `Handle` function. However, the handle function (`log_events.go`), requires us to print out the event in this format:
```
--------------------------- Ansible Task StdOut -------------------------------

 TASK [Debug with loop] ********************************

-------------------------------------------------------------------------------
```
And so the output would become
```
--------------------------- Ansible Task StdOut -------------------------------

 TASK [Debug with loop] ********************************
ok: [localhost] => (item=1) => {
    "msg": 1
}

-------------------------------------------------------------------------------

--------------------------- Ansible Task StdOut -------------------------------

 TASK [Debug with loop] ********************************
ok: [localhost] => (item=2) => {
    "msg": 2
}

-------------------------------------------------------------------------------
```

Which is not what we want, right? Is there any other workaround?

Comment (VenkatRamaraju) [reply internal/ansible/controller/reconcile.go]: Okay, I moved all this logic to `log_events.go` by passing an additional parameter.

**Review by fabianvf — APPROVED** [review]
/lgtm

**Review by varshaprasad96 — APPROVED** [review]

**Pull request merged**